specs/signing-and-verification-workflow.md

```diff
@@ -8,19 +8,19 @@ The user wants to sign an OCI artifact and push the signature to a repository.
 
 ### Signing Prerequisites
 
-- User has access to the signing certificate and private key or a remote signing service through a notation plug-in.
+- User has access to the signing certificate and private key or a remote signing service through a notation plugin.
 
 ### Signing Steps
 
 1. **Generate signature:** Using notation CLI or any other compliant signing tool, sign the OCI artifact. The signing tool should follow the following guideline.
     1. Verify that the signing certificate is valid and satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
     1. Verify that the signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
     1. Generate signature.
-        1. Generate signature using signature formats specified in [supported signature envelopes](./signature-specification.md#supported-signature-envelopes). Also, as part of this step, the user-defined/supplied custom attributes should be added to the annotations of the signature's descriptor.
-        1. If the user wants to timestamp the signature, obtain an [RFC-3161](https://datatracker.ietf.org/doc/html/rfc3161.html) compliant timestamp for the signature generated in the previous step. Otherwise, continue to the next step.
+        1. Generate signature envelope using signature formats specified in [supported signature envelopes](./signature-specification.md#supported-signature-envelopes). Also, as part of this step, the user-defined/supplied custom attributes should be added to the annotations of the signature's descriptor.
+        1. If signing scheme is [`notary.x509`](./signing-scheme.md#notaryx509) and timestamp countersignature is demanded, extract the [primitive signature](./signature-specification.md#signature-envelope) (the digital signature computed on payload and signed attributes) from the signature envelope generated in the previous step. Compute hash of the signature (the hash algorithm SHOULD be deduced from signing certificate's public key. See [algorithm requirements](./signature-specification.md#signature-algorithm-requirements)) and send it to a [RFC 3161](https://datatracker.ietf.org/doc/html/rfc3161.html) compliant *Timestamp Authority (TSA)* for timestamping. Otherwise, continue to the next step. 
             1. Verify that the timestamp signing certificate satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
             1. Verify that the timestamp signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
-        1. Embed timestamp to the signature envelope.
+            1. On success, embed the timestamp countersignature into the *Timestamp Signature* unsigned attribute of the signature envelope. If any of the above timestamping step failed, implementations MAY fail the signing process.
```

Comment:
Timestamping and `tsa` should just be a split of trust. `MUST fail the signing process` actually means moving the entire trust source from `ca` to `tsa`. Since this is the Notary Project spec, I think it's kind of overkill? However, we could enforce this in the Notation implementation if this is the best practice. What do you think? @shizhMSFT

Reply:
The reason why it is a `MAY` instead of `MUST` is to cover a scenario of temporary unavailable TSA so that implementation can choose to update the timestamping signature later on.
Given that we have a workaround of not having a timestamp signature when TSA is not available, I think we can change it to `MUST`.

Reply:
Updated.